x-pack/packages/kbn-ecs-data-quality-dashboard/impl/data_quality/data_quality_panel/index_properties/index.tsx

```diff
@@ -0,0 +1,276 @@
+ * 2.0.
+ */
+
+import { EcsFlat } from '@kbn/ecs';
+import type {
+  FlameElementEvent,
+  HeatmapElementEvent,
+  MetricElementEvent,
+  PartitionElementEvent,
+  Theme,
+  WordCloudElementEvent,
+  XYChartElementEvent,
+} from '@elastic/charts';
+import { EuiSpacer, EuiTab, EuiTabs } from '@elastic/eui';
+import numeral from '@elastic/numeral';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { getUnallowedValueRequestItems } from '../allowed_values/helpers';
+import { ErrorEmptyPrompt } from '../error_empty_prompt';
+import {
+  EMPTY_METADATA,
+  getMappingsProperties,
+  getSortedPartitionedFieldMetadata,
+  hasAllDataFetchingCompleted,
+  INCOMPATIBLE_TAB_ID,
+} from './helpers';
+import { EMPTY_STAT } from '../../helpers';
+import { LoadingEmptyPrompt } from '../loading_empty_prompt';
+import { getTabs } from '../tabs/helpers';
+import { getAllIncompatibleMarkdownComments } from '../tabs/incompatible_tab/helpers';
+import * as i18n from './translations';
+import type { EcsMetadata, IlmPhase, PartitionedFieldMetadata, PatternRollup } from '../../types';
+import { useAddToNewCase } from '../../use_add_to_new_case';
+import { useMappings } from '../../use_mappings';
+import { useUnallowedValues } from '../../use_unallowed_values';
+
+const EMPTY_MARKDOWN_COMMENTS: string[] = [];
+
+interface Props {
+  addSuccessToast: (toast: { title: string }) => void;
+  canUserCreateAndReadCases: () => boolean;
+  defaultNumberFormat: string;
+  docsCount: number;
+  getGroupByFieldsOnClick: (
+    elements: Array<
+      | FlameElementEvent
+      | HeatmapElementEvent
+      | MetricElementEvent
+      | PartitionElementEvent
+      | WordCloudElementEvent
+      | XYChartElementEvent
+    >
+  ) => {
+    groupByField0: string;
+    groupByField1: string;
+  };
+  ilmPhase: IlmPhase | undefined;
+  indexName: string;
+  openCreateCaseFlyout: ({
+    comments,
+    headerContent,
+  }: {
+    comments: string[];
+    headerContent?: React.ReactNode;
+  }) => void;
+  pattern: string;
+  patternRollup: PatternRollup | undefined;
+  theme: Theme;
+  updatePatternRollup: (patternRollup: PatternRollup) => void;
+}
+
+const IndexPropertiesComponent: React.FC<Props> = ({
+  addSuccessToast,
+  canUserCreateAndReadCases,
+  defaultNumberFormat,
+  docsCount,
+  getGroupByFieldsOnClick,
+  ilmPhase,
+  indexName,
+  openCreateCaseFlyout,
+  pattern,
+  patternRollup,
+  theme,
+  updatePatternRollup,
+}) => {
+  const formatNumber = useCallback(
+    (value: number | undefined): string =>
+      value != null ? numeral(value).format(defaultNumberFormat) : EMPTY_STAT,
+    [defaultNumberFormat]
+  );
+  const { error: mappingsError, indexes, loading: loadingMappings } = useMappings(indexName);
+
+  const requestItems = useMemo(
+    () =>
+      getUnallowedValueRequestItems({
+        ecsMetadata: EcsFlat as unknown as Record<string, EcsMetadata>,
```

Comment:
Would it be worth it adding `typeof EcsFlat` to the union type for the ecsMetadata field to all of these functions to avoid this casting?